sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,549 @@
+expensive hardware may be elided. It is undefined behavior to specify the
+`balanced` property on `task_sequence` and then to allow the `task_sequence`
+object to be destroyed while there are any `async` invocations for which `get`
+has not been called.
+
+== `task_sequence` Properties
+
+The following code and table describe the properties that can be provided when
+declaring a `task_sequence` object.
+
+[source,c++,linenums]
+----
+namespace sycl::ext::intel::experimental {
+struct balanced_key {
+  using value_t = property_value<balanced_key>;
+};
+
+struct invocation_capacity_key {
+  template <uint32_t Size>
+  using value_t = property_value<invocation_capacity_key,
+    std::integral_constant<uint32_t, Size>>;
+};
+
+struct response_capacity_key {
+  template <uint32_t Size>
+  using value_t = property_value<response_capacity_key,
+    std::integral_constant<uint32_t, Size>>;
+};
+
+inline constexpr balanced_key::value_t balanced;
+
+template <> struct is_property_key<balanced_key> : std::true_type {};
+
+template <typename ReturnT, typename ... ArgsT, ReturnT(&f) (ArgsT...),
+  class propertiesT>
+struct is_property_key_of<balanced_key,
+  task_sequence<f, propertiesT>> : std::true_type {};
+
+template <uint32_t Size>
+inline constexpr invocation_capacity_key::value_t<Size> invocation_capacity;
+
+template <> struct is_property_key<invocation_capacity_key> : std::true_type {};
+
+template <typename ReturnT, typename ... ArgsT, ReturnT(&f) (ArgsT...),
+  class propertiesT>
+struct is_property_key_of<invocation_capacity_key,
+  task_sequence<f, propertiesT>> : std::true_type {};
+
+template <uint32_t Size>
+inline constexpr response_capacity_key::value_t<Size> response_capacity;
+
+template <> struct is_property_key<response_capacity_key> : std::true_type {};
+
+template <typename ReturnT, typename ... ArgsT, ReturnT(&f) (ArgsT...),
+  class propertiesT>
+struct is_property_key_of<response_capacity_key,
+  task_sequence<f, propertiesT>> : std::true_type {};
+
+} // namespace sycl::ext::intel::experimental
+----
+
+--
+[options="header"]
+|===
+| Property | Description
+| balanced | The `balanced` property is a guarantee to the SYCL device compiler
+that the `task_sequence` object will call exactly the same number of `async` s
+and `get` s over the object's lifetime (i.e. before the `task_sequence`
+destructor is invoked).
+
+| invocation_capacity | The `async` invocations are guaranteed to not block
+as long as the number of outstanding invocations are less than or equal to
+`invocation_capacity`. An outstanding invocation is a call to `async` whose
+corresponding result has not yet been retrieved by a call to `get`.
+
+A default value is chosen by the implementation.
+
+| response_capacity | Given a response capacity of `N`, the user is guaranteed
+that at least one outstanding `async` call will make progress as long as the
+number of outstanding invocations is less than or equal to `N`.
```

Comment:
As above, this is not aligned with the SPIR-V spec and is not usefully different from invocation_capacity in its present state.  A definition that matches the SPIR-V spec would be to say that an outstanding async can make progress so long as the number of completed tasks that haven't yet been geted is less than or equal to 'N'.

Reply:
I have reworded this section with a little more than what you mentioned here.